src/transformers/models/pop2piano/modeling_pop2piano.py

```diff
@@ -0,0 +1,1429 @@
+                                                        n_vocab=composer_n_vocab,
+                                                        n_dim=n_dim
+                                                        )
+        self.transformer = Pop2PianoForConditionalGeneration(config)
+        self.transformer.config.bos_token_id = self.config.eos_token_id
+
+        self.post_init()
+
+    def get_encoder(self):
+        return self.transformer.encoder
+
+    def get_decoder(self):
+        return self.transformer.decoder
+
+    def get_input_embeddings(self):
+        return self.transformer.get_input_embeddings()
+
+    def set_input_embeddings(self, value):
+        self.transformer.decoder.embed_tokens = value
+
+    def freeze_encoder(self):
+        """
+        Calling this function will disable the gradient computation for the Whisper encoder so that its parameters will
+        not be updated during training.
+        """
+        self.transformer.encoder._freeze_parameters()
+
+    def freeze_decoder(self):
+        """
+        Calling this function will disable the gradient computation for the Whisper encoder so that its parameters will
+        not be updated during training.
+        """
+        self.transformer.decoder._freeze_parameters()
+
+    def resize_token_embeddings(self, new_num_tokens: int) -> nn.Embedding:
+        new_embeddings = self.transformer.resize_token_embeddings(new_num_tokens)
+        self.config.vocab_size = new_num_tokens
+        return new_embeddings
+
+    def _freeze_parameters(self):
+        for param in self.parameters():
+            param.requires_grad = False
+        self._requires_grad = False
+
+    def forward(self,
+                inputs: BatchFeature,
+                composer="composer1",
+                max_batch_size:int = None,
+                n_bars:int = 2,
+                return_dict:Optional[bool] = False,
+                **kwargs
+                ) -> Union[Tuple[torch.Tensor], Seq2SeqLMOutput] :
+
+        # select composer randomly if not already given
+        composer_to_feature_token = self.config.composer_to_feature_token
+        if composer is None:
+            composer = np.random.choice(list(composer_to_feature_token.keys()), size=1)[0]
+        elif composer not in composer_to_feature_token.keys():
+            raise ValueError(f"Composer not found in list, Please choose from {list(composer_to_feature_token.keys())}")
+
+        n_bars = self.config.dataset.get("n_bars", None) if n_bars is None else n_bars
+        max_batch_size = 64 // n_bars if max_batch_size is None else max_batch_size
+        max_length = self.config.dataset.get("target_length") * max(1, (n_bars // self.config.dataset.get("n_bars")))
+
+        relative_tokens_arr = []
+        # For multiple audios(batch, seq_len, seq_dim)
+        for i, input_embeds in enumerate(inputs["input_features"]):
+            input_embeds = self.spectrogram(input_embeds).transpose(-1, -2)
```

Comment:
I used both `LogMelSpectrogram` and `ConcatEmbeddingToMel` to be loaded in here, since `ConcatEmbeddingToMel` have weights and it would be better to load them with the main model.